ipa-core/src/net/server/mod.rs

```diff
@@ -443,6 +444,13 @@ impl<S> SetClientIdentityFromHeader<S> {
     fn new(inner: S) -> Self {
         Self { inner }
     }
+
+    fn parse_client_id(header_value: &HeaderValue) -> Result<ClientIdentity, Error> {
```

Comment:
```suggestion
    fn try_parse_client_id(header_value: &HeaderValue) -> Result<ClientIdentity, Error> {
```

Reply:
also why it is in this impl - there is nothing that associated this function with the struct it is defined on. `ClientIdentity` is probably the better place for it